Creiwry/irl-meeting-app
# Fix geolocation on mobile

### Ticket Link
[geolocation](https://templatethpprojetfinal.atlassian.net/browse/THPPROJ-47?atlOrigin=eyJpIjoiZmY0NmM0Y2VlMmVhNDQ1NjhmNzM4ZGMwMDRmMGEzZDgiLCJwIjoiaiJ9)

### User Story
[updating index](https://templatethpprojetfinal.atlassian.net/browse/THPPROJ-12?atlOrigin=eyJpIjoiMGEyMDJlOTJiZGE4NDNjZjkyZjdkMGVjYTllMjUwYzkiLCJwIjoiaiJ9)

### What does this PR do?
It adds a geolocation functionality:
- a geolocation.js file 
   - uses the HTML watchLocation method to watch the user's location
   - uses a geofence to only update the user's location if they move more than 20 meters
- an updade_location
   - in the users controller with route /update_location 
   - receives location params from a request form geolocation.js
   - updates the user's location in the database with these params
- configure an ssl server for localhost for better testing
  - in order to use, it is necessary to generate ssl certificate
  - key.pem  and cert.pem are added to gitignore
!! Adds access to js console in mobile, however, this needs to be removed before release. Only for testing purposes

### Where should the reviewer start?
The reviewer should start in geolocation.js, then in users_controller in the update_location method. Unfortunately, the tests for the js file are not written, so it has to be tested manually.

### How should this be manually tested?
Push local testing branch to heroku
open heroku app on phone
log in as a user
walk around exceeding the distance of 20 meters
check heroku logs --tail for calls to update_location

### Are there any breaking changes?
No
